cmd/queue/main.go

```diff
@@ -133,26 +134,49 @@ var (
 		stats.UnitMilliseconds)
 )
 
-func initEnv() {
-	containerConcurrency = util.MustParseIntEnvOrFatal("CONTAINER_CONCURRENCY", logger)
-	queueServingPort = util.MustParseIntEnvOrFatal("QUEUE_SERVING_PORT", logger)
-	revisionTimeoutSeconds = util.MustParseIntEnvOrFatal("REVISION_TIMEOUT_SECONDS", logger)
-	servingConfig = util.GetRequiredEnvOrFatal("SERVING_CONFIGURATION", logger)
-	servingNamespace = util.GetRequiredEnvOrFatal("SERVING_NAMESPACE", logger)
-	servingPodIP = util.GetRequiredEnvOrFatal("SERVING_POD_IP", logger)
-	servingPodName = util.GetRequiredEnvOrFatal("SERVING_POD", logger)
-	servingRevision = util.GetRequiredEnvOrFatal("SERVING_REVISION", logger)
-	servingService = os.Getenv("SERVING_SERVICE") // KService is optional
-	userTargetPort = util.MustParseIntEnvOrFatal("USER_PORT", logger)
+type envConfig struct {
+	ContainerConcurrency   string `envconfig:"CONTAINER_CONCURRENCY" required:"true"`
+	QueueServingPort       string `envconfig:"QUEUE_SERVING_PORT" required:"true"`
+	RevisionTimeoutSeconds string `envconfig:"REVISION_TIMEOUT_SECONDS" required:"true"`
+	ServingConfiguration   string `envconfig:"SERVING_CONFIGURATION" required:"true"`
+	ServingNamespace       string `envconfig:"SERVING_NAMESPACE" required:"true"`
+	ServingPodIP           string `envconfig:"SERVING_POD_IP" required:"true"`
+	ServingPod             string `envconfig:"SERVING_POD" required:"true"`
+	ServingRevision        string `envconfig:"SERVING_REVISION" required:"true"`
+	// optional
+	ServingService    string `envconfig:"SERVING_SERVICE"`
+	UserPort          string `envconfig:"USER_PORT" required:"true"`
+	UserContainerName string `envconfig:"USER_CONTAINER_NAME" required:"true"`
+	// optional
+	EnableVarLogCollection       string `envconfig:"ENABLE_VAR_LOG_COLLECTION"`
+	VarLogVolumeName             string `envconfig:"VAR_LOG_VOLUME_NAME" required:"true"`
+	InternalVolumePath           string `envconfig:"INTERNAL_VOLUME_PATH" required:"true"`
+	ServingLoggingConfig         string `envconfig:"SERVING_LOGGING_CONFIG" required:"true"`
+	ServingLoggingLevel          string `envconfig:"SERVING_LOGGING_LEVEL" required:"true"`
+	ServingRequestMetricsBackend string `envconfig:"SERVING_REQUEST_METRICS_BACKEND" required:"true"`
+	ServingRequestLogTemplate    string `envconfig:"SERVING_REQUEST_LOG_TEMPLATE" required:"true"`
+}
+
+func initEnv(env envConfig) {
+	containerConcurrency = util.MustParseIntEnvOrFatal("CONTAINER_CONCURRENCY", env.ContainerConcurrency, logger)
+	queueServingPort = util.MustParseIntEnvOrFatal("QUEUE_SERVING_PORT", env.QueueServingPort, logger)
+	revisionTimeoutSeconds = util.MustParseIntEnvOrFatal("REVISION_TIMEOUT_SECONDS", env.RevisionTimeoutSeconds, logger)
```

Comment:
Make all the "MustParseInt" values `int` in the struct above. The library will take care of it!

Reply:
Updated